### ISSUE TITLE
Don't carry out partitioning tasks until we've downloaded all packages successfully

### ISSUE THREAD

**Issue by lots0logs** [Antergos/Cnchi#361]
I'm opening this to remind myself to look into this further. As of now I am not even sure its possible. The following items must be considered:

 - [ ] Can it be done without hacky code.
 - [ ] If partitioning fails, we need to give the user an easy way to recover or at least to save the packages that we've downloaded.
- [ ] Expanding upon the previous point, we should make the installation stateful, meaning it can pick up where it left off. This may require sending the state to the build server for storage (b/c the iso is not persistent storage). 

Looking at this now, I think its probably two separate issues but I'll leave it as one for now. Mostly for brainstorming.

**Comment by axaxs** [comment]
I don't see how we can do this unless we write to ram, which doesn't seem
reliable especially on low memory systems.  What are your thoughts?

**Comment by lots0logs** [comment]
Yeah, like I said, I'm not sure if its possible or feasible. I just thought I should document it while it was on my mind otherwise who knows when I would think about it again :grimacing:

The only reason we would want to do this is because it sucks to re-partition your drive 10 times because your installation keeps failing on the package downloads. Perhaps we can implement some form of statefulness so that Cnchi will skip partitioning if its has already been done. I don't know. What do you guys think?

**Comment by karasu** [comment]
>Perhaps we can implement some form of statefulness so that Cnchi will skip partitioning if its has >already been done

@lots0logs , you can already do this. Simply choose advanced and then choose which partitions you want to use. You won't have to repartition, but you'll have to format the root partition, though.

**Comment by lots0logs** [comment]
That's true, though its not something a regular user will instantly know how to do. But since you mentioned it, I've been thinking about removing the requirement to format root in advanced mode. We could simply delete everything found on it  without formatting. Thoughts?

**Comment by karasu** [comment]
>We could simply delete everything found on it without formatting. Thoughts? 

mmmm.... but then Cnchi will fail if the user doesn't mark the format check on it the first time... don't know, let me think about it.

**Comment by lots0logs** [comment]
Perhaps we could display an alert message stating that the option should only be used after a failed attempt to install where the drive was already partitioned.
